### PR TITLE
Add a "container friendly" log4j2 cfg and output compose dspace log to console

### PR DESCRIPTION
**Pull request by mark-cooper**
## Description

When running dspace in docker it's more convenient to have the logging go to console. To that end this PR provides a variant of the log4j2 cfg that replaces the file with console based logging and overrides the logging config to use the variant in the root level docker compose file.

## Instructions for Reviewers

```bash
# pull the images
docker-compose -f docker-compose.yml -f docker-compose-cli.yml pull

# run the containers
docker-compose -p d7 up # foreground

# or, run in background and tail logs
docker-compose -p d7 up -d
docker logs -f dspace
```

Confirm you're seeing the DSpace logs without having to access the container.

## Checklist

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [ ] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [ ] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [ ] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [ ] If my PR includes new libraries/dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [ ] If my PR modifies REST API endpoints, I've opened a separate [REST Contract](https://github.com/DSpace/RestContract/blob/main/README.md) PR related to this change.
- [ ] If my PR includes new configurations, I've provided basic technical documentation in the PR itself.
- [ ] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

### CODE REVIEW

**Comment by tdonohue** [comment]
@mark-cooper : Absolutely *no* rush on this.  However, if you ever find time to get back to this PR, I think it'd make a great addition to DSpace.  It'd also likely be helpful for the new demo site.   That said, this is much lower priority than anything else...just wanted to mention that this (draft) PR has my interest. :)

**Comment by hardyoyo** [comment]
TLDR: the merge conflict is from the recently merged #8974 and should be very straightforward to resolve. Here's what the conflict looks like:

```
<<<<<<< docker-compose-logcfg
      LOGGING_CONFIG: /dspace/config/log4j2-container.xml
    image: "${DOCKER_OWNER:-dspace}/dspace:${DSPACE_VER:-dspace-7_x-test}"
=======
    image: "${DOCKER_OWNER:-dspace}/dspace:${DSPACE_VER:-latest-test}"
>>>>>>> main
```

**Review by tdonohue — APPROVED** [review]
👍 @mark-cooper :  I resolved the merge conflict & gave this a test today.  It's working for me with my usual process of doing:
```
docker-compose -p d7 up -d
docker-compose -p d7 logs -f
```

I'm now able to see the DSpace logs in the output from compose.

Was there a reason you left this as a "Draft" PR?  It seems like it could be merged as-is, unless there was more work you intended to do to it?

**Comment by mark-cooper** [comment]
@tdonohue iirc at the time I was going to ask if this was wanted for DSpace before pushing it as an in-your-face- PR 😄 -- then got sidetracked in other directions and didn't get back to it. So, if you're happy to accept it then that's great!

**Comment by hardyoyo** [comment]
Tested and approved, I'm merging this.
